RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc

```diff
@@ -106,8 +138,8 @@ void BeamSpotOnlineProducer::produce(edm::Event& iEvent, const edm::EventSetup&
   bool fallBackToDB{false};
 
   // Use transient record if enabled
-  if (useTransientRecord_) {
-    fallBackToDB = processTransientRecord(iSetup, *result, shoutMODE);
+  if (useBSOnlineRecords_) {
+    fallBackToDB = processRecords(iEvent, iSetup, *result, shoutMODE);
```

Comment:
Should the arbitration be done 1 time per lumisection, instead of 1 time per event ? (or, have you observed that the latter is not significantly slower than the former ?)

Reply:
If desired, the arbitration could also be done only when either `BeamSpotOnlineLegacyObjectsRcd` or `BeamSpotOnlineHLTObjectsRcd` (or both) have transitioned to their next IOV.

Reply:
In principle the checks need to be repeated for each lumisection, because the records may become stale (older than `timeThreshold` hours) at any moment.

Reply:
Note that I don't really like that we cannot do these checks in an `ESProducer`, and we have to implement them in an `EDProducer`.